src/components/fx/hover.js

```diff
@@ -549,7 +549,7 @@ function _hover(gd, evt, subplot, noHoverEvent) {
         var thisSpikeDistance;
         for(var i = 0; i < pointsData.length; i++) {
             thisSpikeDistance = pointsData[i].spikeDistance;
-            if(thisSpikeDistance < minDistance && thisSpikeDistance <= spikedistance) {
+            if(thisSpikeDistance <= minDistance && thisSpikeDistance <= spikedistance) {
```

Comment:
The problem stems from `thisSpikeDistance` being `Infinity` and `minDistance` also being `Infinity`. Without an equal sign, this will never be true and no spikelines will appear and hence the bug.

Reply:
Are there any cases this changes *which* point we get spikes to? It might still be OK in that case, I just want to know what those cases are. The loop indicates this way if we have multiple items in `pointData` all with the same `spikeDistance`, previously we'd get the first one, but with the new logic we'll get the last one.

Reply:
> Are there any cases this changes which point we get spikes to? 

@alexcjohnson I'm not sure how to proceed to find out.

Reply:
Should we check whether `minDistance` is `Infinity` and handle that case separately?

Reply:
That's a much better solution, IMO.

Reply:
```js
if((minDistance === Infinity || thisSpikeDistance < minDistance) &&
    thisSpikeDistance <= spikedistance
) {
```
looks like the right logic to apply.

Reply:
Ok so I went with this safer approach. If `spikedistance` is `Infinity` we visit that `if` block. It should preserve any existing behavior but fix the edge case for `Infinity`.

See https://github.com/plotly/plotly.js/pull/4627/commits/583ad647bea336db2c7b844991a94624d762239b

Reply:
The key is to first identify the relevant test cases. I suspect the relevant ones are (with `hovermode: 'x'`):
- Two scatter traces with exactly identical points, so there aren't closer and farther ones. Maybe we don't care which one gets the spike line... though offhand I'd suggest the last one, since it'll be drawn on top so you'll see a spike line color matching the visible point.
- Stacked bar traces with the cursor at a larger (or smaller) y value than any of the bars. Ideally I'd think it would be nice to draw the spike line to the one that's closest; but that would require getting away from using `Infinity` for all points outside the bar, at least for [`sizeFn`](https://github.com/plotly/plotly.js/blob/8f049fddbac0ca0382816984b8526857e9714fe6/src/traces/bar/hover.js#L104). Perhaps we give `Fx.inbox` another (optional) argument `failVal` that, in this case, will be set very large, and increasing the farther away you get from the bar either above or below, but not infinite. Or, at that point it seems silly to use `Fx.inbox` at all, just inline this logic.

Reply:
Thanks for the clarification @alexcjohnson . I now better understand the purpose of this code. I will take another approach.